False positive: ``unnecessary lambda`` with kwargs in the call

### Bug description
```python
# pylint: disable=missing-module-docstring,undefined-variable,missing-function-docstring
def f(a, d):
    print(lambda x: a(x, **d))   
```


### Configuration

_No response_

### Command used

```shell
pylint test.py
```


### Pylint output

```python
************* Module test
test.py:3:2: W0108: Lambda may not be necessary (unnecessary-lambda)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```


### Expected behavior


--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)


### Pylint version

```shell
pylint 3.0.1
astroid 3.0.0
Python 3.11.4 (main, Jun  9 2023, 07:59:55) [GCC 12.3.0]
```


### OS / Environment

Distributor ID: Ubuntu
Description:    Ubuntu 23.04
Release:        23.04
Codename:       lunar

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/9148.false_positive
+++ b/doc/whatsnew/fragments/9148.false_positive
@@ -1,0 +1,3 @@
+Fixed false positive for ``unnecessary-lambda`` when the call has keyword arguments but not the lambda.
+
+Closes #9148

--- a/pylint/checkers/base/basic_checker.py
+++ b/pylint/checkers/base/basic_checker.py
@@ -519,7 +519,6 @@ class BasicChecker(_BasicChecker):
         )
 
     @utils.only_required_for_messages("unnecessary-lambda")
-    # pylint: disable-next=too-many-return-statements
     def visit_lambda(self, node: nodes.Lambda) -> None:
         """Check whether the lambda is suspicious."""
         # if the body of the lambda is a call expression with the same
@@ -544,28 +543,19 @@ class BasicChecker(_BasicChecker):
             # return something else (but we don't check that, yet).
             return
 
-        call_site = astroid.arguments.CallSite.from_call(call)
         ordinary_args = list(node.args.args)
         new_call_args = list(self._filter_vararg(node, call.args))
         if node.args.kwarg:
-            if self._has_variadic_argument(call.kwargs, node.args.kwarg):
+            if self._has_variadic_argument(call.keywords, node.args.kwarg):
                 return
+        elif call.keywords:
+            return
 
         if node.args.vararg:
             if self._has_variadic_argument(call.starargs, node.args.vararg):
                 return
         elif call.starargs:
             return
-
-        if call.keywords:
-            # Look for additional keyword arguments that are not part
-            # of the lambda's signature
-            lambda_kwargs = {keyword.name for keyword in node.args.defaults}
-            if len(lambda_kwargs) != len(call_site.keyword_arguments):
-                # Different lengths, so probably not identical
-                return
-            if set(call_site.keyword_arguments).difference(lambda_kwargs):
-                return
 
         # The "ordinary" arguments must be in a correspondence such that:
         # ordinary_args[i].name == call.args[i].name.

--- a/tests/functional/u/unnecessary/unnecessary_lambda.py
+++ b/tests/functional/u/unnecessary/unnecessary_lambda.py
@@ -52,6 +52,10 @@ _ = lambda: _ANYARGS(**{'three': 3})
 _ = lambda: _ANYARGS(*[3], **{'three': 3})
 _ = lambda: _ANYARGS(func=42)
 
+# pylint: disable=missing-function-docstring
+def f(d):
+    print(lambda x: str(x, **d))
+
 # Don't warn about this.
 _ = lambda: code().analysis()
 

